src/App.js

```diff
@@ -1,19 +1,75 @@
 import React from 'react';
 import './App.css';
+import todosFromServer from './api/todos';
+import { users } from './api/users';
+import { TodoList } from './components/TodoList';
+import { NewTodo } from './components/NewTodo';
 
-import users from './api/users';
+export class App extends React.Component {
+  state = {
+    todos: todosFromServer,
+    todoTitle: '',
+    userName: 'Select a user',
+    titleError: false,
+    userError: false,
+  }
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+  handleTitleChange = (event) => {
+    this.setState({
+      todoTitle: event.target.value.replace(/[^\s\w]/g, ''),
+      titleError: false,
+    });
+  }
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+  handleUserChange = (event) => {
+    this.setState({
+      userName: event.target.value,
+      userError: false,
+    });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    if (this.state.userName === 'Select a user') {
+      this.setState(prevState => ({
+        userError: true,
+      }));
+    } else if (this.state.todoTitle === '') {
+      this.setState(prevState => ({
+        titleError: true,
+      }));
```

Comment:
prevstate же не используешь, значит можно без него
```suggestion
    if (this.state.userName === 'Select a user') {
      this.setState(({
        userError: true,
      });
    } else if (this.state.todoTitle === '') {
      this.setState({
        titleError: true,
      });
```

Reply:
Так, дякую) Це я копіював і не помітив